File: src/main/java/org/jhapy/audit/domain/Session.java
package org.jhapy.audit.domain;

import java.time.Instant;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.ToString;
import org.springframework.data.mongodb.core.index.Indexed;
import org.springframework.data.mongodb.core.mapping.Document;

/**
 * @author jHapy Lead Dev.
 * @version 1.0
 * @since 2019-05-24
 */
@Document(collection = "session")
@Data
@EqualsAndHashCode(callSuper = true)
@ToString(callSuper = true)
public class Session extends BaseEntity {

  private String username;

  private String sourceIp;

  private Instant sessionStart;

  private Instant sessionEnd;

  private Long sessionDuration;

  private Boolean isSuccess;

  private String error;

  @Indexed(unique = true)
  private String jsessionId;

}

File: src/main/java/org/jhapy/audit/domain/Session.kt
/*
 * Copyright 2020-2020 the original author or authors from the JHapy project.
 *
 * This file is part of the JHapy project, see https://www.jhapy.org/ for more information.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *       http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package org.jhapy.audit.domain

import org.springframework.data.mongodb.core.index.Indexed
import org.springframework.data.mongodb.core.mapping.Document
import java.time.Instant

/**
 * @author jHapy Lead Dev.
 * @version 1.0
 * @since 2019-05-24
 */
@Document(collection = "session")
class Session : BaseEntity() {
    var username: String? = null
    var sourceIp: String? = null
    var sessionStart: Instant? = null
    var sessionEnd: Instant? = null
    var sessionDuration: Long? = null
    var isSuccess: Boolean? = null
    var error: String? = null

    @Indexed(unique = true)
    var jsessionId: String? = null
}